### PR TITLE
Bug 1635148 - Recent changes to add support for attachments in S3 caused a memory leak in the phabbugz feed daemon

### PR DESCRIPTION
**Pull request by dklawren**

### CODE REVIEW

File: Bugzilla/Attachment/Storage/S3.pm [hunk 1]
```diff
@@ -38,15 +45,15 @@ sub data_type { return 's3'; }
...
 
   # If the attachment is larger than attachment_s3_minsize,
   # we instead store it in the database.
   if (Bugzilla->params->{attachment_s3_minsize}
-    && $self->attachment->datasize < Bugzilla->params->{attachment_s3_minsize})
+    && $self->datasize < Bugzilla->params->{attachment_s3_minsize})
   {
     require Bugzilla::Attachment::Storage::Database;
-    return Bugzilla::Attachment::Storage::Database->new({attachment => $self->attachment})
+    return Bugzilla::Attachment::Storage::Database->new({attach_id => $self->attach_id, datasize => $self->datasize})
```
Comment (globau) [inline Bugzilla/Attachment/Storage/S3.pm:56]: `Bugzilla::Attachment::Storage::Database` doesn't accept a `datasize` parameter

File: extensions/PhabBugz/lib/Feed.pm [hunk 2]
```diff
@@ -537,7 +539,7 @@ sub process_revision_change {
   # Email changes for this revisions bug and also for any other
   # bugs that previously had these revision attachments
   foreach my $bug_id ($revision->bug_id, keys %other_bugs) {
-    Bugzilla::BugMail::Send($bug_id, {changer => $changer->bugzilla_user});
+   Bugzilla::BugMail::Send($bug_id, {changer => $changer->bugzilla_user});
```
Comment (globau) [inline extensions/PhabBugz/lib/Feed.pm:542]: the indentation was correct before

File: extensions/PhabBugz/lib/Feed.pm [hunk 3]
```diff
@@ -483,6 +483,8 @@ sub process_revision_change {
     }
 
     $attachment->update($timestamp);
+
+    undef $attachment;
```
Comment (globau) [inline extensions/PhabBugz/lib/Feed.pm:487]: this shouldn't be required, but doesn't hurt either.

Comment (dklawren) [reply extensions/PhabBugz/lib/Feed.pm:487]: Yeah that was from some previous troubleshooting and was not removed. Will remove.

File: extensions/PhabBugz/lib/Feed.pm (hunk 2 above)
Comment (dklawren) [reply extensions/PhabBugz/lib/Feed.pm:542]: Fixing

File: Bugzilla/Attachment/Storage/S3.pm (hunk 1 above)
Comment (dklawren) [reply Bugzilla/Attachment/Storage/S3.pm:56]: Fixing

**Pull request merged**